`parserOptions.project` doesnt handle project references well

**What code were you trying to parse?**

Not parsing code exactly, but I'm attempting to upgrade to v2, which hard fails if `project` doesn't include files correctly. This doesn't seem to handle project references correctly.

Given this root `tsconfig.json`:

```
{
  "files": [],
  "references": [
    {
      "path": "packages/config-danger"
    }
  ],
  "extends": "./tsconfig.options.json"
}
```

And this `config-danger/tsconfig.json`:

```
{
  "compilerOptions": {
    "declarationDir": "lib",
    "outDir": "lib",
    "rootDir": "src"
  },
  "exclude": [
    "lib",
    "tests"
  ],
  "extends": "../../tsconfig.options.json",
  "include": [
    "src/**/*",
    "types/**/*",
    "../../types/**/*"
  ],
  "references": []
}
```

**What did you expect to happen?**

Files within a project reference `include` worked correctly.

**What actually happened?**

The parser/plugin fails with a ton of the following errors:

```
/root/packages/config-danger/src/code.ts
  0:0  error  Parsing error: If "parserOptions.project" has been set for @typescript-eslint/parser, /root/packages/config-danger/src/code.ts must be included in at least one of the projects provided
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.0.0` |
| `TypeScript`                | `3.5.2` |
| `ESLint`                    | `6.1.0` |
| `node`                      | `12.8.0` |
| `npm`                       | `6.1.0.3` |

--- a/packages/parser/README.md
+++ b/packages/parser/README.md
@@ -66,9 +66,11 @@ The following additional configuration options are available by specifying them 
     ];
     ```
 
+  - Note that if you use project references, TypeScript will not automatically use project references to resolve files. This means that you will have to add each referenced tsconfig to the `project` field either separately, or via a glob.
+
   - Note that if this setting is specified and `createDefaultProgram` is not, you must only lint files that are included in the projects as defined by the provided `tsconfig.json` files. If your existing configuration does not include all of the files you would like to lint, you can create a separate `tsconfig.eslint.json` as follows:
 
-    ```js
+    ```jsonc
     {
       // extend your base config so you don't have to redefine your compilerOptions
       "extends": "./tsconfig.json",

